great_expectations/cli/datasource.py

```diff
@@ -1220,117 +1220,79 @@ def _get_batch_kwargs_from_generator_or_from_file_path(
     if datasource.engine.dialect.name.lower() == "snowflake":
         # snowflake requires special handling
-        table_name = click.prompt(
+        snowflake_transient_table = click.prompt(
             "In Snowflake, GE may need to create a transient table "
             "to use for validation."
             + os.linesep
             + "Please enter a name to use for that table: ",
             default="ge_tmp_" + str(uuid.uuid4())[:8],
         )
         temp_table_kwargs = {
-            "snowflake_transient_table": table_name,
+            "snowflake_transient_table": snowflake_transient_table,
         }
     elif datasource.engine.dialect.name.lower() == "bigquery":
         # bigquery also requires special handling
-        table_name = click.prompt(
+        bigquery_temp_table = click.prompt(
             "GE will create a table to use for "
             "validation." + os.linesep + "Please enter a name for this table: ",
             default="SOME_PROJECT.SOME_DATASET.ge_tmp_" + str(uuid.uuid4())[:8],
         )
         temp_table_kwargs = {
-            "bigquery_temp_table": table_name,
+            "bigquery_temp_table": bigquery_temp_table,
         }
 
-    while True:
-        try:
-            query = None
-
-            if len(available_data_asset_names) > 0:
-                selection = click.prompt(prompt, show_default=False)
-
-                selection = selection.strip()
-                try:
-                    data_asset_index = int(selection) - 1
-                    try:
-                        data_asset_name = [
-                            name[0] for name in available_data_asset_names
-                        ][data_asset_index]
-                    except IndexError:
-                        pass
-                except ValueError:
-                    query = selection
-
-            else:
-                query = click.prompt(msg_prompt_query, show_default=False)
-
-            if query is None:
-                batch_kwargs = temp_generator.build_batch_kwargs(
-                    data_asset_name, **additional_batch_kwargs
-                )
-                batch_kwargs.update(temp_table_kwargs)
-            else:
-                batch_kwargs = {"query": query, "datasource": datasource_name}
-                batch_kwargs.update(temp_table_kwargs)
-                Validator(
-                    batch=datasource.get_batch(batch_kwargs),
-                    expectation_suite=ExpectationSuite("throwaway"),
-                ).get_dataset()
-
-            break
-        except ge_exceptions.GreatExpectationsError as error:
-            cli_message("""<red>ERROR: {}</red>""".format(str(error)))
-        except KeyError as error:
-            cli_message("""<red>ERROR: {}</red>""".format(str(error)))
+    try:
+        if sql_query is None:
+            # construct the query explicitly using schema_name and table_name
+            sql_query = f"select * from {schema_name}.{table_name}"
```

Comment:
This change converts *all* table batch_kwargs to query batch_kwargs. This is not desirable, since it will force creating temp tables in a lot of cases when they are not needed. I'd suggest to go back to allow table bath kwargs when user provides a table name.

Reply:
fixed via 696987b